### PR TITLE
Preserve legacy mount ordering.

### PR DESCRIPTION
**Pull request by calavera**
To remain backwards compatible:

Volume -> Bind Mount -> Volumes-From

Signed-off-by: David Calavera david.calavera@gmail.com

### CODE REVIEW

**Comment by cpuguy83** [comment]
LGTM

**Comment by tiborvass** [comment]
@calavera looks great, but i wanna make sure this testcase is there as well:
```
$ echo world > somepath/hello
$ docker run --name c1 -v somepath:/foo busybox true
$ docker run --volumes-from c1 busybox cat /foo/hello
world
```

**Comment by calavera** [comment]
@tiborvass I've added another test.

File: integration-cli/docker_cli_run_test.go [hunk 1]
```diff
@@ -550,6 +550,34 @@ func (s *DockerSuite) TestRunMultipleVolumesFrom(c *check.C) {
 	}
 }
 
+func (s *DockerSuite) TestRunVolumesFromAfterBind(c *check.C) {
```
Comment (cpuguy83) [inline integration-cli/docker_cli_run_test.go]: This will require `SameHostDaemon` check.

**Comment by calavera** [comment]
There was some confusion here and @tiborvass and I decided to not port this back. This is a small corner case that we don't want to encourage. Bind mounts should always override any other mounts in the same destination path.

We want this:

Image Volume -> Volumes From -> Bind Mount

Not this:

Image Volume -> Bind Mount -> Volumes From

Because bind mounts are the ones that the final user actually has control over.

**Pull request closed without merging**